File: core/WiggleSortII/WiggleSortII.java
package com.dbc.code;

import java.util.Arrays;

public class WiggleSortII {
    public void wiggleSort(int[] nums) {
        int[] arr = nums.clone();
        Arrays.sort(arr);
        int x = (nums.length + 1) / 2 - 1;
        int k = nums.length - 1;
        for (int i = 0; i < nums.length; i += 2) {
            nums[i] = arr[x];
            if (i + 1 < nums.length) {
                nums[i + 1] = arr[k];
            }
            x--;
            k--;
        }
    }
}

File: core/WiggleSortII/WiggleSortII.py
import math
from typing import List


def wiggleSort(nums: List[int]) -> None:
    n = len(nums)
    arr = sorted(nums)
    x = (n + 1) // 2
    j, k = x - 1, n - 1
    for i in range(0, n, 2):
        nums[i] = arr[j]
        if i + 1 < n:
            nums[i + 1] = arr[k]
        j -= 1
        k -= 1


if __name__ == '__main__':
    print(wiggleSort([1,3,2,2,3,1]))




